src/test/java/io/appium/java_client/localserver/StartingAppLocally.java

```diff
@@ -0,0 +1,122 @@
+package io.appium.java_client.localserver;
+
+import io.appium.java_client.MobileElement;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.ios.IOSDriver;
+import io.appium.java_client.remote.AutomationName;
+import io.appium.java_client.remote.MobileCapabilityType;
+import io.appium.java_client.remote.MobilePlatform;
+import io.appium.java_client.service.local.AppiumServiceBuilder;
+import io.appium.java_client.service.local.flags.GeneralServerFlag;
+import org.junit.Assert;
+import org.junit.Test;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+
+public class StartingAppLocally {
+
+    @Test
+    public void startingAndroidAppWithCapabilitiesOnlyTest(){
+        File appDir = new File("src/test/java/io/appium/java_client");
+        File app = new File(appDir, "ApiDemos-debug.apk");
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android Emulator");
+        capabilities.setCapability(MobileCapabilityType.APP, app.getAbsolutePath());
+        capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.APPIUM);
+
+        AndroidDriver<MobileElement> driver = null;
+        try {
+            driver = new AndroidDriver<>(capabilities);
+            Capabilities caps = driver.getCapabilities();
+
+            Assert.assertEquals(true, caps.getCapability(MobileCapabilityType.AUTOMATION_NAME).equals(AutomationName.APPIUM));
+            Assert.assertEquals(true, caps.getCapability(MobileCapabilityType.PLATFORM_NAME).equals(MobilePlatform.ANDROID));
+            Assert.assertNotEquals(null, caps.getCapability(MobileCapabilityType.DEVICE_NAME));
+            Assert.assertEquals(true, caps.getCapability(MobileCapabilityType.APP).equals(app.getAbsolutePath()));
+        }
+        finally {
+            if (driver != null)
```

Comment:
Have you looked at Guava's [Optional](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Optional.html) class? It's similar to [Optional in JDK 8](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) except works with older JDKs.